server/server.go

```diff
@@ -443,8 +443,17 @@ func (s *Server) printConfiguration() {
 	for res != nil {
 		logger().Infof("-> resolver: '%s'", resolver.Name(res))
 
-		for _, c := range res.Configuration() {
-			logger().Infof("     %s", c)
+		if resCfg, ok := res.(resolver.ConfigGetter); ok {
+			func() {
+				undo := log.PrefixMessages("     ", logger().Logger)
+				defer undo()
```

Comment:
Is it possible to create a new logger instance with prefix formatter? I mean a new instance only for configuration logging. I think, it is a bit "dangerous" to modify the existing instance and to rollback this change after usage. Later someone can use this logic somewhere else and forget to call the undo function.

Reply:
Yeah I can do that too, this was just a bit easier with the existing `log` code!

Reply:
So I tried changing this, but using a new logger loses stuff like fields that were added.
I changed the API to be impossible to misuse though, so hopefully that solves the issue with another approach!